### PR TITLE
Refactor voucher tests file

### PR DESCRIPTION
**Pull request by zguesmi**
* Use global variables to remove duplication.
* Create common used voucher only once in `beforeEach` block.
* Create mock orders in `beforeEach` block.

### CODE REVIEW

File: test/beacon/Voucher.test.ts [hunk 1]
```diff
@@ -48,26 +77,51 @@ describe('Voucher', function () {
...
         ] = await ethers.getSigners();
-        const beacon = await voucherUtils.deployBeaconAndImplementation(admin.address);
+        // Deploy PoCo mock and VoucherHub.
         iexecPocoInstance = await new IexecPocoMock__factory()
             .connect(admin)
             .deploy()
             .then((x) => x.waitForDeployment());
         iexecPoco = await iexecPocoInstance.getAddress();
-        const voucherHub = await voucherHubUtils.deployHub(
+        beacon = await voucherUtils.deployBeaconAndImplementation(admin.address);
```
Comment (gfournieriExec) [inline test/beacon/Voucher.test.ts:86]: Why move this line here and not like before on top of line 81 => will remove some diff for the PR stat (?)

File: test/beacon/Voucher.test.ts [hunk 2]
```diff
@@ -48,26 +77,51 @@ describe('Voucher', function () {
...
-        voucherHubWithAnyoneSigner = voucherHub.connect(anyone);
-        await voucherHubWithAssetEligibilityManagerSigner.createVoucherType(description, duration);
-
+        // Fund VoucherHub with RLCs.
         await iexecPocoInstance
             .transfer(await voucherHub.getAddress(), initVoucherHubBalance)
             .then((tx) => tx.wait());
+        // TODO rename to voucherHubAs...
+        voucherHubAsVoucherCreationManager = voucherHub.connect(voucherManager);
+        voucherHubAsAssetEligibilityManager = voucherHub.connect(assetEligibilityManager);
```
Comment (gfournieriExec) [inline test/beacon/Voucher.test.ts:99]: Same here => move on top of line 94 ?

File: test/beacon/Voucher.test.ts [hunk 3]
```diff
@@ -83,38 +137,22 @@ describe('Voucher', function () {
 
     describe('Upgrade', function () {
         it('Should upgrade all vouchers', async function () {
-            const { beacon, voucherHub, admin, voucherOwner1, voucherOwner2 } =
-                await loadFixture(deployFixture);
-            const voucherType1 = 1;
-            const duration1 = 7200;
-            const description1 = 'Long Term Duration';
-            const voucherValue1 = 200;
```
Comment (gfournieriExec) [inline test/beacon/Voucher.test.ts:91]: The point was not to have a different duration of the common one ?

File: test/beacon/Voucher.test.ts (hunk 1 above)
Comment (zguesmi) [reply test/beacon/Voucher.test.ts:86]: The goal is to separate different parts:
* First, deploy PoCo
* Then, deploy Voucher contracts.
* Fund VoucherHub
* ...etc
Also, this PR stats are quite good so I don't think it's needed here.

File: test/beacon/Voucher.test.ts (hunk 2 above)
Comment (zguesmi) [reply test/beacon/Voucher.test.ts]: I agree, it makes more sense to init all VoucherHub instances at once.
[Create different VoucherHub instances at the same place](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/18/commits/b0ed7e02aabf7b5880a9153ba71388b0883d2ed3)

File: test/beacon/Voucher.test.ts (hunk 3 above)
Comment (zguesmi) [reply test/beacon/Voucher.test.ts:91]: Good catch!
[Create voucher with different type and value](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/18/commits/295037d0de82487ac453f31bced067556e48c21c)

**Review by gfournieriExec — APPROVED** [review]
Thanks for the cleaning !

**Pull request merged**